packages-exp/auth-exp/src/platform_browser/iframe/iframe.test.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+
+import { SDK_VERSION } from '@firebase/app-exp';
+import { FirebaseError } from '@firebase/util';
+
+import { TEST_AUTH_DOMAIN, TEST_KEY, testAuth } from '../../../test/mock_auth';
+import { stubSingleTimeout } from '../../../test/timeout_stub';
+import { Auth } from '../../model/auth';
+import { AUTH_WINDOW } from '../auth_window';
+import * as gapiLoader from './gapi';
+import { _openIframe } from './iframe';
+
+use(sinonChai);
+use(chaiAsPromised);
+
+type IframesCallback = (iframesLib: unknown) => Promise<unknown>;
+
+describe('src/platform_browser/iframe/iframe', () => {
+  let auth: Auth;
+  let iframeSettings: Record<string, unknown>;
+  let libraryLoadedCallback: IframesCallback;
+
+  beforeEach(async () => {
+    AUTH_WINDOW.gapi = ({
+      iframes: {
+        CROSS_ORIGIN_IFRAMES_FILTER: 'cross-origin-filter'
+      }
+    } as unknown) as typeof gapi;
```

Comment:
why do we need these recasts?

Reply:
because otherwise we'd have to create a type-valid mock of all the properties we define on gapi, when the code under test uses a small subset of them. And we have to go through `unknown` first because the type system thinks it's sooo clever